kotlin/export-tiles/src/main/java/com/example/exporttiles/MainActivity.kt

```diff
@@ -0,0 +1,263 @@
+          // convert screen points to map points
+          val minPoint: Point = mapView.screenToLocation(minScreenPoint)
+          val maxPoint: Point = mapView.screenToLocation(maxScreenPoint)
+          // use the points to define and return an envelope
+          downloadArea?.geometry = Envelope(minPoint, maxPoint)
+        }
+      }
+    }
+
+    // when the button is clicked, export the tiles to the temporary directory
+    exportTilesButton.setOnClickListener {
+      val exportTileCacheTask = ExportTileCacheTask(tiledLayer.uri)
+      // set up the export tile cache parameters
+      val parametersFuture: ListenableFuture<ExportTileCacheParameters> =
+        exportTileCacheTask.createDefaultExportTileCacheParametersAsync(
+          downloadArea?.geometry,
+          mapView.mapScale,
+          tiledLayer.maxScale
+        )
+
+      parametersFuture.addDoneListener {
+        try {
+          val parameters: ExportTileCacheParameters = parametersFuture.get()
+          // create a temporary directory in the app's cache for saving exported tiles
+          val exportTilesDirectory = File(cacheDir, getString(R.string.tile_cache_folder))
+          // export tiles to temporary cache on device
+          exportTileCacheJob =
+            exportTileCacheTask.exportTileCache(
+              parameters,
+              exportTilesDirectory.path + "file.tpk"
+            ).apply {
+              // start the export tile cache job
+              start()
+
+              // show progress of the export tile cache job on the progress bar
+              val dialog = createProgressDialog(this)
+              dialog.show()
+
+              // on progress change
+              addProgressChangedListener {
+                dialog.progressBar.progress = progress
+                dialog.progressTextView.text = "$progress%"
+              }
+
+              // when the job has completed, close the dialog and show the job result in the map preview
+              addJobDoneListener {
+                dialog.dismiss()
+                if (status == Job.Status.SUCCEEDED) {
+                  showMapPreview(result)
+                  downloadArea?.isVisible = false
+
+                } else {
+                  ("Job did not succeed: " + error.additionalMessage).also {
+                    Toast.makeText(this@MainActivity, it, Toast.LENGTH_LONG).show()
+                    Log.e(TAG, error.additionalMessage)
+                  }
+                }
+              }
+            }
+        } catch (e: InterruptedException) {
+          Toast.makeText(this, "TileCacheParameters interrupted: " + e.message, Toast.LENGTH_LONG)
+            .show()
+          Log.e(TAG, "TileCacheParameters interrupted: " + e.message)
+        } catch (e: ExecutionException) {
+          Toast.makeText(this, "Error generating parameters: " + e.message, Toast.LENGTH_LONG)
+            .show()
+          Log.e(TAG, "Error generating parameters: " + e.message)
+        }
```

Comment:
```suggestion
          catch (e: Exception) {
          val error = "Error generating tile cache parameters: ${e.message}"
          Toast.makeText(this, error, Toast.LENGTH_LONG).show()
          Log.e(TAG, error)
        } 
```

Reply:
Since the difference between interrupted vs Execution exceptions aren't too important. Perhaps its a bit easier to read as above?